src/main/webapp/common_util.js

```diff
@@ -13,16 +13,18 @@
 // limitations under the License.
 
 /*
- * Extracts video id from full url
- */ 
-function cleanseUrl(url) {
-  // Split web address from parameters, extract first parameter
-  // TODO: Add checks to make this work if video is not first parameter.
-  var videoId = url.split("?");
-  videoId = (videoId.length > 1) ? videoId[1].split("&")[0] : videoId[0];
-
-  // If param name present, remove it to isolate video Id.
-  videoId = videoId.replace("v=", "");
-  
-  return videoId;
-}
\ No newline at end of file
+ * @param url: url of the video to be analyzed
+ * @returns: just the video id of that video
+ * Credit for RegEx method derived from top answer on: 
+ * https://stackoverflow.com/questions/28735459/how-to-validate-youtube-url-in-client-side-in-text-box
+ */	 
+function extractYouTubeUrl(url) {
+  const idLength = 11; // Length of video Id's in YouTube videos
+  if (url != undefined || url != '') {
+    // This regular expression represents the different patterns that may occur in a YouTube URL
```

Comment:
As the linter suggests can you make it so these lines are only 80 characters long?

Reply:
I don't think I can do this without breaking up the regExp String, which would honestly look worse and maybe even cause an error, but I should do that for the comment.